learn/core_concepts/documents.md

```diff
@@ -38,6 +46,40 @@ In the above example, `"id"`, `"title"`, `"genre"`, `"release-year"`, and `"cast
 Each attribute must be associated with a **value**, e.g. `"Kung Fu Panda"` is the value of `"title"`.
 At minimum, the document must contain one field with the **[primary key][primary-key]** attribute and a unique **[document id][document-id]** as its value. Above, that's: `"id": "1564saqw12ss"`.
 
+#### NDJSON
+
+NDJSON objects consist of individual lines where each individual line is valid JSON text and each line is delimited with a newline character. Any [rules that apply to formatting NDJSON](http://ndjson.org/) also apply to MeiliSearch documents.
+
+Compared to JSON, NDJSON has better writing performance and is less CPU and memory intensive. It is easier to validate and can handle nested structures.  
+
+The above JSON document would look like this in NDJSON:
+
+```json
+
+{ 
+  "id": "1564saqw12ss", 
+  "title": "Kung Fu Panda", 
+  "genre": "Children's Animation", 
+  "release-year": 2008, 
+  "cast": [ {"Jack Black": "Po"}, {"Jackie Chan": "Monkey"} ]
+  }
+```
+
+#### CSV
+
+CSV files express data as a sequence of values separated by a delimiter character. Though the comma is the most common delimiter, spaces, tabs, and semicolons are valid alternatives. Any [rules that apply to formatting CSV](https://datatracker.ietf.org/doc/html/rfc4180) also apply to MeiliSearch documents.
+
+Compared to JSON, CSV has better writing performance and is less CPU and memory intensive.  
+
+The above JSON document would look like this in CSV:
+
+```csv
+  "id:string","title:string","genre:string","release-year:number"
```

Comment:
If `:string` or `:number` is not specified, it will fallback to `:string` by default. I think it's worth mentioning 😇

Reply:
Updated to add this as tip